rpc/cmd.go

```diff
@@ -131,14 +142,15 @@ func (CmdRsSetUserVoiceSettings) messageData() {}
 
 type CmdArgsSelectVoiceChannel struct {
 	ChannelID snowflake.ID `json:"channel_id"`
-	Timeout   int          `json:"timeout"`
+	Timeout   int          `json:"timeout,omitempty"`
 	Force     bool         `json:"force"`
+	Navigate  bool         `json:"navigate"`
 }
 
 func (CmdArgsSelectVoiceChannel) cmdArgs() {}
 
 type CmdRsSelectVoiceChannel struct {
-	PartialChannel
+	*PartialChannel
```

Comment:
is this a pointer because the result can be null?

Reply:
Correct. When leaving a voice channel, the result is null.

Reply:
then you wont need this to be a pointer
```suggestion
	PartialChannel
```

Reply:
for `CmdRsSelectVoiceChannel` that makes sense. What about `CmdRsGetSelectedVoiceChannel`? Should that remain a pointer (`if channel == nil` to detect no channel) or should I also remove the pointer there?  (`if channel.ID == 0` to detect no channel)

Reply:
that shouldnt be a pointer either
the struct will be nil in this case

Reply:
structs can't be `nil` though. it defaults to a struct with all 0 values

Reply:
ah right we don't use a pointer here https://github.com/dennisrijsdijk/disgo/blob/feature/rpc-rewrite/rpc/message.go#L188
then this is fine and we can check for `id == 0`

Reply:
Then again, I'm working on the earlier comment that everything should return `(*T, error)` so I could add that logic into the function there as well.

Reply:
sounds good to me

Reply:
Resolved in [4ef8ffc](https://github.com/dennisrijsdijk/disgo/commit/4ef8ffc03ba58f70cb22951e155ddf66e18ef4e6)